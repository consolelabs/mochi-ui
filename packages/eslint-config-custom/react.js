const { resolve } = require('node:path')

const project = resolve(process.cwd(), 'tsconfig.json')

const dfLintConfig = require.resolve('@dwarvesf/react-eslint-config')

module.exports = {
  extends: [dfLintConfig, 'eslint-config-turbo'],
  plugins: ['prettier'],
  parserOptions: {
    project,
  },
  globals: {
    JSX: true,
  },
  ignorePatterns: [
    'node_modules/',
    'dist/',
    '.eslintrc.js',
    '**/*.css',
    '*.snap',
  ],
  settings: {
    'import/resolver': {
      typescript: {
        project,
      },
      node: {
        extensions: ['.mjs', '.js', '.jsx', '.ts', '.tsx'],
      },
    },
  },
  rules: {
    'react/function-component-definition': 'off',
    'turbo/no-undeclared-env-vars': 'off',
    'react/jsx-key': 'off',
    'no-dupe-class-members': 'off',
    'prefer-arrow-callback': 'off',
    'import/export': 'off',
    'prettier/prettier': [
      'error',
      {
        bracketSpacing: true,
        jsxBracketSameLine: false,
        jsxSingleQuote: false,
        printWidth: 80,
        proseWrap: 'always',
        semi: false,
        singleQuote: true,
        tabWidth: 2,
        trailingComma: 'all',
        xmlWhitespaceSensitivity: 'ignore',
      },
    ],
  },
  overrides: [
    {
      parser: '@typescript-eslint/parser',
      files: ['**/*.ts', '**/*.tsx'],
      plugins: ['@typescript-eslint'],
      rules: {
        'no-use-before-define': 'off',
        'react/react-in-jsx-scope': 'off',
        'react/jsx-sort-props': 'off',
        'no-void': 'off',
        'no-unused-vars': 'off',
        '@typescript-eslint/no-unused-vars': [
          'error',
          {
            argsIgnorePattern: '^_',
            varsIgnorePattern: '^_',
            caughtErrorsIgnorePattern: '^_',
          },
        ],
        '@typescript-eslint/no-use-before-define': [
          'error',
          { functions: false, classes: false },
        ],
      },
    },
  ],
}
