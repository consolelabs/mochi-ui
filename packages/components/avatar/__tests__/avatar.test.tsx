import { render } from '@testing-library/react'
import Avatar from '../src/avatar'

describe('Avatar', () => {
  it('renders correctly with src', () => {
    const { container } = render(
      <Avatar size="base" src="https://example.com/avatar.png" />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders correctly with smallSrc', () => {
    const { container } = render(
      <Avatar
        size="sm"
        src="https://example.com/avatar.png"
        smallSrc="https://example.com/small-avatar.png"
      />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders correctly with fallback', () => {
    const { container } = render(
      <Avatar
        size="base"
        fallback="fallback"
        src="https://example.com/avatar.png"
      />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
