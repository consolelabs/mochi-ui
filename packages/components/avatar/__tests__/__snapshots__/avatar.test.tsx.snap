// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly with fallback 1`] = `
<span
  class="flex-shrink-0 rounded-full w-10 h-10 overflow-hidden"
>
  <span>
    <img
      alt="fallback"
      src="https://source.boringavatars.com/beam/120/fallback?colors=665c52,74b3a7,a3ccaf,E6E1CF,CC5B14"
    />
  </span>
</span>
`;

exports[`Avatar renders correctly with smallSrc 1`] = `
<div
  class="flex-shrink-0 rounded-full w-7 h-7"
>
  <svg
    height="100%"
    role="none"
    viewBox="0 0 100 100"
    width="100%"
  >
    <mask
      id="circle-mask-:r1:"
    >
      <circle
        cx="50%"
        cy="50%"
        fill="white"
        r="50%"
      />
      <circle
        cx="80%"
        cy="80%"
        fill="black"
        r="24%"
      />
    </mask>
    <image
      class="overflow-hidden"
      height="100%"
      mask="url(#circle-mask-:r1:)"
      width="100%"
      xlink:href="https://example.com/avatar.png"
    />
    <image
      height="40%"
      width="40%"
      x="60%"
      xlink:href="https://example.com/small-avatar.png"
      y="60%"
    />
  </svg>
</div>
`;

exports[`Avatar renders correctly with src 1`] = `
<span
  class="flex-shrink-0 rounded-full w-10 h-10 overflow-hidden"
>
  <span>
    <img
      alt="fallback"
      src="https://source.boringavatars.com/beam/120/?colors=665c52,74b3a7,a3ccaf,E6E1CF,CC5B14"
    />
  </span>
</span>
`;
